docs(web-analytics): Update cookieless and cookie banner tutorials (#12584)
PostHog/posthog.com: contents/tutorials/react-cookie-banner.md

Next, we’ll go to `src/main.jsx` to initialize PostHog. We’ll import PostHog and the `PostHogProvider` component from `posthog-js/react` and run `posthog.init` with our project key and instance address (which you can get in your [project settings](https://us.posthog.com/settings/project))

> ```js

Reviewer: ```suggestion
```js file=main.jsx
```

we can use this, too.

    </PostHogProvider>
  </React.StrictMode>,
)
```

Once done, visit your site running locally, click the button, and you should start seeing events in your [PostHog activity tab](https://us.posthog.com/activity/explore).

<ProductScreenshot
  imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/Clean_Shot_2024_07_10_at_14_54_53_2x_de2d28eea9.png" 
  imageDark="https://res.cloudinary.com/dmukukwp6/image/upload/Clean_Shot_2024_07_10_at_14_54_41_2x_13e7233065.png"
  alt="Events in PostHog" 
  classes="rounded"
/>

When we head back to our local site, right-click, choose inspect, go to the Application tab, and check cookies, you’ll see we’ve created a tracking cookie with details about PostHog and the user. 

![App cookie](https://res.cloudinary.com/dmukukwp6/image/upload/Clean_Shot_2024_07_10_at_14_59_04_2x_19bbe1f339.png)

## Step 3: Ensuring cookies aren't set on initial load

> To start making this more compliant, we want to ensure we don't store any cookies until the user opts in. 

Reviewer: Nit, but maybe we just say "making this fully compliant" instead of "more compliant"

I think more/less compliant isn't the right wording here. I _think_ you are either compliant or you aren't. Ignore me if you have ideas otherwise :)

To do this, we set `cookieless_mode` to `on_reject` in our initialization config like this:

> ```js focusOnLines=8-12

Reviewer: Nice!

// main.jsx
import React from 'react'
import ReactDOM from 'react-dom/client'
import App from './App.jsx'
import './index.css'
import posthog from 'posthog-js'
import { PostHogProvider } from 'posthog-js/react'

posthog.init("<ph_project_api_key>", {
  api_host: "<ph_client_api_host>",
  defaults: "<ph_posthog_js_defaults>",
  cookieless_mode: 'on_reject'
})

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <PostHogProvider client={posthog}>
      <App />
    </PostHogProvider>
  </React.StrictMode>,
)
```

This means that PostHog will not set any cookies until the user has given consent, which is what we rely on the cookie banner to do.

## Step 4: Building the banner component

Now that we have our site and PostHog set up, it is time to create the banner. To start, we can create a `Banner.jsx` file in our `src` folder (where the rest of our React code is).

```bash
cd src
touch Banner.jsx
```

> In the `Banner.jsx` file, we’ll create a basic React component function with wording about cookies and accept or decline buttons connected to `handleAcceptCookies` and `handleDeclineCookies` functions (that just log to the console for now).

Reviewer: ```suggestion
In the `Banner.jsx` file, we’ll create a basic React component with a blurb about cookies. In this banner, we'll add accept and decline buttons connected to the `handleAcceptCookies` and `handleDeclineCookies` functions (that just log to the console for now).
```

}
```

Back on our site, clicking accept or decline now opts users in or out of tracking cookies and hides the banner.

## Step 6: Testing it all out

To make sure everything works, try the following steps:

1. Go to the site and click accept or decline on the banner.

> 2. Check the banner disappears and whether cookies are set or not.

Reviewer: ```suggestion
2. Confirm that the banner disappears and whether cookies are set or not.
```
